# Brazil Covid-19 Data Analysis

#### Author: Romullo Ferreira

The aim of this project is to analyze Covid-19 (Brazil) data, which includes the following steps:

1. Defining the list of questions to be answered with data analysis.
2. Data wrangling 
    - 2.1. Gather
    - 2.2. Assess
    - 2.3. Clean
3. Data exploration and effective communication of the exploratory steps.
4. Conclusions.

Useful facts about the data:
- epidemiological situation of COVID-19 in Brazil
- The data update process is carried out daily by the Ministry of Health through official information provided by the State Health Secretariats of the 27 Brazilian Federative Units
- 27 states of Brazil
- 5 regions of brazil (Central-West, North, Northeast, South and Southeast)
- Data were collected from 2021-03-09 to 2021-03-18
- Data Source: State Health Secretariats. Brazil, 2020
- Detailed data can be downloaded from [OpenDataSUS](https://opendatasus.saude.gov.br/)
- You can also access the [Brazilian Open Data Portal](https://dados.gov.br/dataset) to find more coronavirus data

## 1. Defining the list of questions to be answered with data analysis

We have a lot of things to ask. The dataset has a lot of information, but for the analysis not to be too long I decided to ask some questions that I thought were important as a start. But after taking a look at my analysis feel free to ask other questions and analyze.

It is important to remember that All questions below refer to Covid-19 data from Brazil:

- a. - What is the total accumulated cases?
- b. - What is the total new cases?
- c. - What is the total recovered cases?
- d. - What is the total followup cases?
- e. - What is the total Covid-19 deaths?
- f. - What is the total new deaths?
- g. - What is the Incidence Coefficient of Covid-19?
- h. - What is the Mortality Coefficient for Covid-19?
- i. - What is the Covid-19 Lethality Rate?
- j. - Which region(zone) has the highest Covid-19 Cases?
- k. - Which states have the highest number of covid cases?
- l. - Which cities have the highest number of covid cases?
- m. - Which region(zone) has the highest Covid-19 deaths?
- n. - Which states have the highest number of covid deaths?
- o. - Which cities have the highest number of covid deaths?

## 2. Data wrangling

## 2.1. Gather

Firstly, let's import all the libraries necessary for this project

In [3]:
import pandas as pd
import numpy as np

import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

/usr/local/lib/python3.6/dist-packages/matplotlib/__init__.py:886: MatplotlibDeprecationWarning: 
examples.directory is deprecated; in the future, examples will be found relative to the 'datapath' directory.
  "found relative to the 'datapath' directory.".format(key))


Import CSV File

In [4]:
df_covid19br = pd.read_csv('COVIDBR.csv', sep=';')

## 2.2. Assess

Let's take a look at the dataframe

In [5]:
df_covid19br.sample(3)

regiao estado   municipio  coduf    codmun  codRegiaoSaude  \
45540      Norte     AM    Eirunepé     13  130140.0         13007.0   
418060  Nordeste     RN      Jaçanã     24  240500.0         24005.0   
497409  Nordeste     PB  Mãe d'Água     25  250870.0         25006.0   

                        nomeRegiaoSaude        data  semanaEpi  \
45540                    REGIONAL JURUA  2020-12-30         53   
418060  5ª REGIAO DE SAUDE - SANTA CRUZ  2020-07-25         30   
497409                        6ª REGIAO  2021-03-13         10   

        populacaoTCU2019  casosAcumulado  casosNovos  obitosAcumulado  \
45540            35273.0            2709           0               12   
418060            9133.0             168           0                1   
497409            4009.0             129           1                4   

        obitosNovos  Recuperadosnovos  emAcompanhamentoNovos  \
45540             0               NaN                    NaN   
418060            0               NaN                    NaN   
497409            0               NaN                    NaN   

        interior/metropolitana  
45540                      0.0  
418060                     0.0  
497409                     0.0

We can already see that there are missing values in the columns estado, municipio, codmun and other columns for example.

Here we can see the size of original dataset, 2.012.472 rows e 17 columns.

In [6]:
df_covid19br.shape

(2012472, 17)

Let's count the unique values

In [7]:
df_covid19br.nunique()

regiao                        6
estado                       27
municipio                  5297
coduf                        28
codmun                     5591
codRegiaoSaude              450
nomeRegiaoSaude             440
data                        389
semanaEpi                    53
populacaoTCU2019           5104
casosAcumulado            33078
casosNovos                 3779
obitosAcumulado            7273
obitosNovos                 641
Recuperadosnovos            336
emAcompanhamentoNovos       336
interior/metropolitana        2
dtype: int64

##### Let's create a copy of the original dataframe. And create another dataframe (filtered_df) with just today's date data, which is where I'm going to focus my analysis.

In [8]:
filtered_df = df_covid19br.copy()
filtered_df = filtered_df.query("data > '2021-03-18'")
filtered_df.sample(3)

regiao estado   municipio  coduf    codmun  codRegiaoSaude  \
800641   Nordeste     BA       Saúde     29  292980.0         29014.0   
1022243   Sudeste     MG  Pedra Azul     31  314870.0         31068.0   
1759365       Sul     RS  Montenegro     43  431240.0         43008.0   

        nomeRegiaoSaude        data  semanaEpi  populacaoTCU2019  \
800641         JACOBINA  2021-03-19         11           12913.0   
1022243      PEDRA AZUL  2021-03-19         11           24324.0   
1759365       REGIAO 08  2021-03-19         11           65264.0   

         casosAcumulado  casosNovos  obitosAcumulado  obitosNovos  \
800641              325           2                2            0   
1022243             936           2               22            0   
1759365            5550         182               76            1   

         Recuperadosnovos  emAcompanhamentoNovos  interior/metropolitana  
800641                NaN                    NaN                     0.0  
1022243               NaN                    NaN                     0.0  
1759365               NaN                    NaN                     1.0

Dimensions of the dataframe. Here we can already see that our filtered dataframe is much smaller than the original file, with 5619 rows and 17 columns. 

In [9]:
filtered_df.shape

(5619, 17)

- Which columns have missing values? 
- Here we can already see that the "munincipio" column has only 5570 against 5619 of the original file. This means that we have 49 missing values. And the "codmun" column has only 5591 out of a total of 5619, this means that we have 28 missing values. Let's keep that information.

In [10]:
filtered_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5619 entries, 390 to 2012471
Data columns (total 17 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   regiao                  5619 non-null   object 
 1   estado                  5618 non-null   object 
 2   municipio               5570 non-null   object 
 3   coduf                   5619 non-null   int64  
 4   codmun                  5591 non-null   float64
 5   codRegiaoSaude          5570 non-null   float64
 6   nomeRegiaoSaude         5570 non-null   object 
 7   data                    5619 non-null   object 
 8   semanaEpi               5619 non-null   int64  
 9   populacaoTCU2019        5598 non-null   float64
 10  casosAcumulado          5619 non-null   int64  
 11  casosNovos              5619 non-null   int64  
 12  obitosAcumulado         5619 non-null   int64  
 13  obitosNovos             5619 non-null   int64  
 14  Recuperadosnovos        1 non-null 

Another way to see missing values. Count of missing values.

In [11]:
filtered_df.isnull().sum()

regiao                       0
estado                       1
municipio                   49
coduf                        0
codmun                      28
codRegiaoSaude              49
nomeRegiaoSaude             49
data                         0
semanaEpi                    0
populacaoTCU2019            21
casosAcumulado               0
casosNovos                   0
obitosAcumulado              0
obitosNovos                  0
Recuperadosnovos          5618
emAcompanhamentoNovos     5618
interior/metropolitana      49
dtype: int64